### ISSUE TITLE
Add reference to generated resource file

### ISSUE THREAD

**Issue by dexpota** [tuist/tuist#5543]
### What problem or need do you have?

I have a project where one resource file is generated after the Xcode project is created with tuist. In this scenario, tuist does not obviously add this file. I tried to add the path to this generated file to the array of resources inside the project definition. In this case, tuist issues a warning but does not fail.

```swift
Target(
    name: "App",
    platform: .iOS,
    product: .app,
    bundleId: "com.example.app",
    infoPlist: .file(path: "App/Info.plist"),
    sources: ["App/**"],
    resources: [
        "App/Resource/This-is-a-generated-file.plist"
    ],
    dependencies: []
)
```

The problem is that the generated project does not have a reference to this resource file, and the application compiles but crashes at runtime because this file is missing. I'd prefer to catch this missing file at compile time and not at runtime.

I could solve this issue by simply moving the file generation before the project generation, but this is problematic because of two reasons. First, it creates an implicit dependency between steps inside the CI/CD that is not clearly visible by looking at the code. Second, if the generation of this file fails for some reason, it wouldn't be possible to catch this problem until I try the application.

These are my first steps with tuist and iOS projects and maybe I am missing something. I am open to suggestions. Thank you.

### macOS version

14.0

### Tuist version

3.30.0

### Xcode version

15.0

**Comment by danyf90** [comment]
Yep, you should generate it before the project is generated, it's strange to me that if it doesn't find it it's not a linting error 🤔

**Comment by dexpota** [comment]
I know the solution is to generate the file beforehand, but I am not comfortable with the idea to have the possibility to generate a project that compiles but fails at runtime.

tuist issues a warning but it does generate the project, this is the output

```
Resolved cache profile 'Development' from Tuist's defaults
No files found at: <project-root>/App/Resource/This-is-a-generated-file.plist
Generating workspace AppWorkspace.xcworkspace
Generating project App
Project generated.
Total time taken: 1.690s
```

**Comment by fortmarek** [comment]
I think we should keep this a warning. That being said, maybe there's a space for `tuist generate --strict` to turn warnings to errors. It has been discussed before [here](https://github.com/tuist/tuist/issues/4376). Back then we decided it's good enough to print warnings at the end, so they are more visible. But I can still see value of the `--strict` mode, especially for the CI.

**Comment by dexpota** [comment]
I think this `--strict` flag could be useful, especially in this scenario where I don't want the project to be generated if the file is missing.
